File: src/data/models/CreditCard.ts
import { DocumentModel } from "./DocumentModel";

export class CreditCard extends DocumentModel {
  constructor(
    public id: string,
    public name: string,
    public limit: number,
    public brand: string,
    public accountId: string,
    public closingDay: number,
    public dueDay: number,
    public archived: boolean = false,
    public importInfo?: any
  ) { 
    super(id);
  }
}

File: Android/app/src/main/kotlin/br/com/victorwads/goldenunicorn/data/models/CreditCard.kt
package br.com.victorwads.goldenunicorn.data.models

import com.google.firebase.firestore.DocumentId
import com.google.firebase.firestore.PropertyName

data class CreditCard(
    @DocumentId val id: String = "",
    @PropertyName("name") val name: String = "",
    @PropertyName("limit") val limit: Double = 0.0,
    @PropertyName("brand") val brand: String = "",
    @PropertyName("accountId") val accountId: String = "",
    @PropertyName("closingDay") val closingDay: Int = 1,
    @PropertyName("dueDay") val dueDay: Int = 1,
    @PropertyName("archived") val archived: Any? = null,
)
